sqlparser/lib/src/analysis/steps/prepare_ast.dart

```diff
@@ -170,16 +170,22 @@ class AstPreparingVisitor extends RecursiveVisitor<void, void> {
     visitChildren(e, arg);
   }
 
-  void _resolveIndexOfVariables() {
+  @override
+  void visitNestedQueryVariable(NestedQueryVariable e, void arg) {
+    _foundVariables.add(e);
+    visitChildren(e, arg);
```

Comment:
If we end up keeping the `NestedQueryVariable` in sqlparser, it sounds like this is essentially unreachable since we only add them in a later analysis phase? Or am I misreading something?

Reply:
No, I think that’s right. I probably only added this for completeness sake, but it could be removed.

Reply:
The `assert(false)` is a very good idea for that, thanks!